### ISSUE TITLE
sundials build error

### ISSUE THREAD

**Issue by balay** [xsdk-project/xsdk-issues#215]
sundials/trilinos dependency issues?

[spack-build-out.txt](https://github.com/xsdk-project/xsdk-issues/files/12253171/spack-build-out.txt)

ref: ./bin/spack install -j32 xsdk@1.0.0

**Comment by csiefer2** [comment]
Sundails devs: Karen Devine's talk from the Trilinos Users Group Meeting 2021 can likely help with the transition to Trilinos 14+ syntax in Tpetra:

https://trilinos.github.io/pages/community/trilinos_user_meetings/trilinos_user-developer_group_meeting_2021/2021-11-30/04.2-Devine-2021_TUG_Tpetra.pdf

**Comment by balay** [comment]
Attaching the current log
[spack-build-out.txt](https://github.com/xsdk-project/xsdk-issues/files/12363105/spack-build-out.txt)

**Comment by balay** [comment]
My prior log is with sundials/develop branch.

However 6.6.0 [already released] is set as the verison for xsdk@1.0.0 - so switching to this version also results in a build failure

[spack-build-out.txt](https://github.com/xsdk-project/xsdk-issues/files/12371802/spack-build-out.txt)

**Comment by gardner48** [comment]
I've opened LLNL/sundials#326 to add support for Trilinos 14+. @balay could you try testing with the `feature/trilinos-14` branch and let me know if that resolves the build issues?

**Comment by gardner48** [comment]
@balay LLNL/sundials#326 was merged. Can you try testing with `@develop` and let me know if that resolves the build problems?

**Comment by gardner48** [comment]
> @balay [LLNL/sundials#326](https://github.com/LLNL/sundials/pull/326) was merged.

Oops, was getting my PRs confused. This has not been merged yet, so `feature/trilinos-14` is still the branch to test.

**Comment by balay** [comment]
>  feature/trilinos-14 is still the branch to test.

@gardner48, the build goes through fine now. Thanks!

```
+    version("6.6.1", branch="feature/trilinos-14")
```
```
==> Installing sundials-6.6.1-3riezzg3ytjzlfh35xtkbh3ozpf6lvfd [94/101]
==> No binary for sundials-6.6.1-3riezzg3ytjzlfh35xtkbh3ozpf6lvfd found: installing from source
==> No patches needed for sundials
==> sundials: Executing phase: 'cmake'
==> sundials: Executing phase: 'build'
==> sundials: Executing phase: 'install'
==> sundials: Successfully installed sundials-6.6.1-3riezzg3ytjzlfh35xtkbh3ozpf6lvfd
  Stage: 5.86s.  Cmake: 11.21s.  Build: 16.45s.  Install: 1.12s.  Post-install: 0.29s.  Total: 35.26s
[+] /data/balay/spack.x/opt/spack/linux-rocky9-cascadelake/gcc-11.3.1/sundials-6.6.1-3riezzg3ytjzlfh35xtkbh3ozpf6lvfd
```

**Comment by gardner48** [comment]
> the build goes through fine now.

Thanks for testing this out @balay. The PR has been merged to `@develop` now so you can use that for xsdk testing until we get a new release out.

**Comment by balay** [comment]
Thanks! switched xsdk@1.0.0 testing back to `develop` branch

**Comment by gardner48** [comment]
> Thanks! switched xsdk@1.0.0 testing back to `develop` branch

@balay We've released `sundials@6.6.1` with this fixed and added this version to spack

**Comment by balay** [comment]
Thanks! will give that a try
